### ISSUE TITLE
Review HyperV enlightenments for Windows guest

### ISSUE THREAD

**Issue by rbradford** [cloud-hypervisor/cloud-hypervisor#4779]
We only currently enable a subset of HyperV enlightenments to support running Windows guests: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/f32487f8e8958be7e7b9e696caa484383fc570bb/arch/src/x86_64/mod.rs#L675

I think this list could be broadened to improve the Windows guest experience.

**Comment by liuw** [comment]
I think Vitaly's talk has a lot of useful information.

https://www.youtube.com/watch?v=ZYlaDqwL9Jw&ab_channel=TheLinuxFoundation

**Comment by weltling** [comment]
Would possibly an upgrade to the Windows guest version be relevant in this scope. Windows Server 2022 is out already, so a corresponding guest image could be added to the CI and might trigger a need on more enlightenments.

Otherwise, we perhaps could create a kind of list to track which features are already enabled, which are not enabled and which might need to be enabled. That list would be the base for a roadmap to follow on any listed Hyper-V enlightenments in question.

Thanks

**Comment by weltling** [comment]
An update here - i went for creating a Windows Server 2022 image and made a basic smoke test with 28.0. It didn't show any obvious issues. There is no automation creating these images, as they're based on the stock ISO from the VS subscription. Nevertheless, got that new guest image now so we could replace the older one server 2019 or add the new one in a parallel run. 

> Perhaps we should be more flexible here. Instead of hiding the enlightenments behind one knob we can expose them for users to decide?

To clarify this one - would this mean user would be able to set any arbitrary bits? If so, it might likely lead to issues as some bits would need additional handling that wouldn't be available? Of course, it would also induce more bug reports that could give info for further considerations.

Thanks

**Comment by rbradford** [comment]
@weltling We only enable a subset of the possible HyperV enlightenments. Enabling some of the others may improve performance without requiring any extra work. QEMU offers fine grained control of the HyperV bits; I don't think that is necessary but i'm sure our default list could be extended.

**Comment by weltling** [comment]
@rbradford absolutely, I've read it same way as you just clarified. My question was related only to the point about "let users to decide", as that can only be done by exposing it through a console option and can possibly lead to some misbehavior. Perhaps it could be done as an experimental feature to make experimenting with the bits easier. Or perhaps we could enable more and expose an option to disable some bits, the other way round so to say. All in all, the existing bits and the info pointed out by Wei are sure a good base to go with enabling more.  

Thanks

**Comment by rbradford** [comment]
> @rbradford absolutely, I've read it same way as you just clarified. My question was related only to the point about "let users to decide", as that can only be done by exposing it through a console option and can possibly lead to some misbehavior. Perhaps it could be done as an experimental feature to make experimenting with the bits easier. Or perhaps we could enable more and expose an option to disable some bits, the other way round so to say. All in all, the existing bits and the info pointed out by Wei are sure a good base to go with enabling more.

I would really rather avoid adding options if we can get away without adding them.

**Comment by rbradford** [comment]
Our Windows support is pretty mature now and i'm not aware of any issues folks are having with this. So I think we can stick with enlightenments we have
